dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -124,13 +131,49 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
         THROW_HR_IF(E_UNEXPECTED, notificationId == 0);
         toast.ToastId(notificationId);
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */, hstring /* group */)
+
+    // API updates notification data when both tag and group are provided or when tag is provided but group is empty.
+    // For other scenarios the API returns ToastProgressResult::Failed.
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData data, hstring tag, hstring group)
     {
-        throw hresult_not_implemented();
+        std::wstring appId{ RetrieveToastAppId() };
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::IToastProgressData> toastProgressData
+            = winrt::make_self<ToastProgressDataABI>(data);
```

Comment:
Ohhhh so close. Got it right just 2 lives above ;-)

`winrt::com_ptr<...> toastProgressData{ winrt::make_self<...>(data); }`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)